src/plugins/explore/public/components/query_panel/query_panel_editor/query_panel_editor.tsx

```diff
@@ -32,7 +32,7 @@ export const QueryPanelEditor = () => {
       data-test-subj="exploreQueryPanelEditor"
       onClick={onEditorClick}
     >
-      <CodeEditor {...editorProps} />
+      <CodeEditor key={editorProps.languageId} {...editorProps} />
```

Comment:
This will cause the whole query editor to rerender. There must be a better way to handle this